ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py

```diff
@@ -529,3 +529,76 @@ def test_torch_svdvals(
         on_device=on_device,
         A=x,
     )
+
+
+# tensorinv
+@st.composite
+def _get_inv_square_matrices(draw):
+    dim_size = draw(helpers.ints(min_value=1, max_value=10))
+
+    batch_shape = draw(st.sampled_from([2, 4, 6, 8, 10]))
+
+    generated_shape = (dim_size,) * batch_shape
+    generated_ind = int(np.floor(len(generated_shape) / 2))
+
+    handpicked_shape, handpicked_ind = draw(
+        st.sampled_from([[(24, 6, 4), 1], [(8, 3, 6, 4), 2], [(6, 7, 8, 16, 21), 3]])
+    )
+
+    shape, ind = draw(
+        st.sampled_from(
+            [(generated_shape, generated_ind), (handpicked_shape, handpicked_ind)]
+        )
+    )
+
+    input_dtype = draw(
+        helpers.get_dtypes("float", index=1, full=False).filter(
+            lambda x: x not in ["float16", "bfloat16"]
+        )
+    )
+    invertible = False
+    while not invertible:
+        input = draw(
+            helpers.array_values(
+                dtype=input_dtype[0],
+                shape=shape,
+                min_value=-100,
+                max_value=100,
+            )
+        )
+        try:
+            np.linalg.inv(input)
+            invertible = True
+        except np.linalg.LinAlgError:
+            pass
+
+    return input_dtype, input, ind
+
+
+@handle_frontend_test(
+    fn_tree="torch.linalg.tensorinv",
+    dtype_input_ind=_get_inv_square_matrices()
+)
+def test_torch_tensorinv(
+        *,
+        dtype_input_ind,
+        as_variable,
+        num_positional_args,
+        native_array,
+        on_device,
+        fn_tree,
+        frontend,
+):
+    dtype, x, ind = dtype_input_ind
+    helpers.test_frontend_function(
+        input_dtypes=dtype,
+        as_variable_flags=as_variable,
+        with_out=False,
```

Comment:
As I said above you'll need to add an `out` argument, so then you should set `with_out=True`.